src/main/java/frc/robot/commands/TeleopDriveCommand.java

```diff
@@ -24,12 +27,67 @@ public class TeleopDriveCommand extends Command {
     private ProfiledPIDController rotationController = new ProfiledPIDController(DriveConstants.kAimP,
             DriveConstants.kAimI, DriveConstants.kAimD, DriveConstants.kAimProfile);
 
-    /** Creates a new TeleopDrive. */
+    /**
+     * Creates a new TeleopDriveCommand Command.
+     * 
+     * @param driveSubsystem   The drive subsystem.
+     * @param xSpeedSupplier   The x speed.
+     * @param ySpeedSupplier   The y speed.
+     * @param rotSpeedSupplier The angular speed.
+     * @param rotationAngle    The rotation angle.
+     */
     public TeleopDriveCommand(DriveSubsystem driveSubsystem,
             DoubleSupplier xSpeedSupplier,
             DoubleSupplier ySpeedSupplier,
             DoubleSupplier rotSpeedSupplier,
             Double rotationAngle) {
+        InitializeTeleopDriveCommand(driveSubsystem, xSpeedSupplier, ySpeedSupplier, rotSpeedSupplier, rotationAngle);
+    }
+
+    /**
+     * Creates a new TeleopDriveCommand Command.
+     * 
+     * @param driveSubsystem   The drive subsystem.
+     * @param xSpeedSupplier   The x speed.
+     * @param ySpeedSupplier   The y speed.
+     * @param rotSpeedSupplier The angular speed.
+     * @param stageSide        LEFT or RIGHT
+     */
+    public TeleopDriveCommand(DriveSubsystem driveSubsystem,
+            DoubleSupplier xSpeedSupplier,
+            DoubleSupplier ySpeedSupplier,
+            DoubleSupplier rotSpeedSupplier,
+            StageSide stageSide) {
+
+        Double rotationAngle = 0.0;
+        Alliance currentAlliance = DriverStation.getAlliance().get();
```

Comment:
The Alliance logic **needs** to move to initialize(), otherwise it will take the Alliance from when the robot code boots. On a real field, it will likely not be connected to the FMS when this command is created, so you will not have the correct alliance.